tests/test_main.py

```diff
@@ -355,3 +359,29 @@ def test_make_summary_without_calling_log(tmp_dir):
     }
     log_file = tmp_dir / dvclive.plots_dir / Metric.subfolder / "foo.tsv"
     assert not log_file.exists()
+
+
+def test_make_summary_is_called_on_end(tmp_dir):
+    live = Live()
+
+    live.summary["foo"] = 1.0
+    live.end()
+
+    assert json.loads((tmp_dir / live.metrics_file).read_text()) == {
+        # no `step`
+        "foo": 1.0
+    }
+    log_file = tmp_dir / live.plots_dir / Metric.subfolder / "foo.tsv"
+    assert not log_file.exists()
+
+
+def test_context_manager(tmp_dir):
+    with Live() as live:
+        live.summary["foo"] = 1.0
```

Comment:
`__exit__` only calls `live.make_summary` and `post_to_studio` for now. 

After https://github.com/iterative/dvc/pull/8408 is available, the context manager should take care of generating the DVC experiment

Reply:
@dberenbaum Should `__exit__` also call `make_report()`? I think so

Reply:
Yup, I agree, it should call `make_report()`.